packages/formik/src/Formik.tsx

```diff
@@ -129,33 +148,45 @@ interface FieldRegistry {
   };
 }
 
-export function useFormik<Values extends FormikValues = FormikValues>({
-  validateOnChange = true,
-  validateOnBlur = true,
-  validateOnMount = false,
-  isInitialValid,
-  enableReinitialize = false,
-  onSubmit,
-  ...rest
-}: FormikConfig<Values>) {
+export function useFormik<Values extends FormikValues = FormikValues>(
+  rawProps: FormikConfig<Values>
+): FormikApi<Values> {
+  const {
+    validateOnChange = true,
+    validateOnBlur = true,
+    validateOnMount = false,
+    enableReinitialize = false,
+    ...rest
+  } = rawProps;
   const props = {
     validateOnChange,
     validateOnBlur,
     validateOnMount,
-    onSubmit,
     ...rest,
   };
-  const initialValues = React.useRef(props.initialValues);
-  const initialErrors = React.useRef(props.initialErrors || emptyErrors);
-  const initialTouched = React.useRef(props.initialTouched || emptyTouched);
-  const initialStatus = React.useRef(props.initialStatus);
+
+  if (__DEV__) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    React.useEffect(() => {
+      invariant(
+        typeof props.isInitialValid === 'undefined',
+        'isInitialValid has been deprecated and will be removed in future versions of Formik. Please use initialErrors or validateOnMount instead.'
+      );
+      // eslint-disable-next-line
+    }, []);
+  }
+
+  /**
+   * Refs
+   */
   const isMounted = React.useRef<boolean>(false);
   const fieldRegistry = React.useRef<FieldRegistry>({});
+
   if (__DEV__) {
     // eslint-disable-next-line react-hooks/rules-of-hooks
     React.useEffect(() => {
       invariant(
-        typeof isInitialValid === 'undefined',
+        typeof props.isInitialValid === 'undefined',
```

Comment:
Should we actually break the rules of hooks here? (i believe the rationale is that we only ever want to show this error message once) If so, let's add a comment about it or fix this.

Reply:
Yeah, I didn't make any changes to this except to update the variable, but I believe you wrote it so that it would only print the warning once in development on the first use.

I would create a separate issue for this since it's unrelated to this PR.

Reply:
Created https://github.com/formium/formik/issues/3145 for this, as I wasn't involved in the __DEV__ checks.